httpe/method.go

```diff
@@ -0,0 +1,37 @@
+package httpe
+
+import "net/http"
+
+// Get returns a ErrMethodNotAllowed if request method is not a GET. Use with Chain.
+var Get = HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error { return ensureMethod(r, http.MethodGet) })
+
+// Head returns a ErrMethodNotAllowed if request method is not a HEAD. Use with Chain.
+var Head = HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error { return ensureMethod(r, http.MethodHead) })
+
+// Post returns a ErrMethodNotAllowed if request method is not a Post. Use with Chain.
+var Post = HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error { return ensureMethod(r, http.MethodPost) })
+
+// Put returns a ErrMethodNotAllowed if request method is not a Put. Use with Chain.
+var Put = HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error { return ensureMethod(r, http.MethodPut) })
+
+// Patch returns a ErrMethodNotAllowed if request method is not a Patch. Use with Chain.
+var Patch = HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error { return ensureMethod(r, http.MethodPatch) })
+
+// Delete returns a ErrMethodNotAllowed if request method is not a Delete. Use with Chain.
+var Delete = HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error { return ensureMethod(r, http.MethodDelete) })
+
+// Connect returns a ErrMethodNotAllowed if request method is not a Connect. Use with Chain.
+var Connect = HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error { return ensureMethod(r, http.MethodConnect) })
+
+// Options returns a ErrMethodNotAllowed if request method is not a Options. Use with Chain.
+var Options = HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error { return ensureMethod(r, http.MethodOptions) })
+
+// Trace returns a ErrMethodNotAllowed if request method is not a Trace. Use with Chain.
+var Trace = HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error { return ensureMethod(r, http.MethodTrace) })
+
+func ensureMethod(r *http.Request, method string) error {
```

Comment:
you could simplify the vars by using a constructor function to create the checkers:
```
var (
    Get = newMethodChecker(http.MethodGet)
    Post = newMethodChecker(http.MethodPost)
...
)

func newMethodChecker(method string) HandlerFuncE {
  return HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) {
    if r.Method != method {
      return ErrMethodNotAllowed
    }
    return nil
  })
}
```

Reply:
dang I feel like I wrote exactly that a little while ago and we decided to drop it
anyway, yes this is better, fixed.
Also the `Any` comment is very valid, however I think if we want to allow routing for more than on method then usually there are different handlers for say a post and a get.

What would be great is to have a handlerE shorthand that does the demultiplexing to the right handlerFuncE function or so, like a muxE or so... not very deeply thought out but somewhat inspired by chi:
```
muxE("api/thing")
      .Post(postThingHandlerE)
      .Get(getThingHandlerE)
 ```

Reply:
agree. we'll save that discussion for another time. routing and error handling should be orthogonal (mostly), so not sure the best way to integrate the two.